### PR TITLE
#6 Task. AlexErm

### PR DESCRIPTION
**Pull request by alexErm0**
https://travis-ci.com/github/alexErm0/js-assignments/builds/175123463

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,13 @@
  *
  */
 function getFizzBuzz(num) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:32]: fix according to google's style guide

https://google.github.io/styleguide/jsguide.html#formatting-nonempty-blocks

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -30,7 +30,13 @@
...
-    throw new Error('Not implemented');
+	if(num%3!=0 && num%5!=0) return num
+	else {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: you don't need else here becuase in above statement you have return in such cases else can be omitted

E.g.
```
if ( condtion ) {
    return value1;
}
return value2;
```

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -46,7 +52,13 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:57]: fix indentation

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -63,7 +75,9 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+	let sum = 0;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: fix indentation

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -82,7 +96,10 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    if(a+b<=c)return false
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: fix indentation

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -150,7 +175,9 @@ function doRectanglesOverlap(rect1, rect2) {
  *   
  */
 function isInsideCircle(circle, point) {
-    throw new Error('Not implemented');
+    let dirLine= Math.sqrt(Math.pow(circle.center.x-point.x, 2) + Math.pow(circle.center.y-point.y, 2));
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: fix indentation
Use Math.hypot

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -150,7 +175,9 @@ function doRectanglesOverlap(rect1, rect2) {
...
+	if(dirLine>=circle.radius) return false
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: just return !(bolean expression)

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -226,7 +268,10 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    let STR = 0;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: local variables must be in lowercase

File: task/06-conditions-n-loops-tasks.js [hunk 9]
```diff
@@ -332,7 +435,18 @@ function isBracketsBalanced(str) {
  *
  */
 function timespanToHumanString(startDate, endDate) {
-    throw new Error('Not implemented');
+    var diffSec= Math.abs(endDate.getTime()-startDate.getTime());
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: don't use vars

**Review by AlexMokin — COMMENTED** [review]
Indentations are not consistent. There are violations of the google's code style.

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Review by AlexMokin — COMMENTED** [review]
Review this https://javascript.info/coding-style . And try to find code styling mistakes yourself and fix them.

Also theere must not be any changes to the formatting of the original code, see image below

image

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js [hunk 10]
```diff
@@ -192,7 +219,19 @@ function findFirstSingleChar(str) {
  *
  */
 function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
-    throw new Error('Not implemented');
+	if(a>b){
+		var a1=b;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not use var. use either let or const

File: task/06-conditions-n-loops-tasks.js [hunk 11]
```diff
@@ -209,7 +248,9 @@ function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
  * 'noon' => 'noon'
  */
 function reverseString(str) {
-    throw new Error('Not implemented');
+	let result = "";
+	for(let i = 0; i<str.length; i++) result += str[str.length-1-i]
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: https://javascript.info/coding-style#curly-braces

File: task/06-conditions-n-loops-tasks.js [hunk 12]
```diff
@@ -226,7 +267,10 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+	let str = 0;
+	let num1 = new String(num);
+	for(let i = 1; i<=num1.length; i++) str += Math.pow(10, num1.length-i)*(Math.floor(num/Math.pow(10, i-1))%10);
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: https://javascript.info/coding-style#line-length
https://javascript.info/coding-style#curly-braces

image

**Review by AlexMokin — COMMENTED** [review]
1) fix this comments
2) original indetations are 4 spaces. Use same indetations for your code
3)  Pull latest changes from master

You have similar mistakes in all branches. I'll mark all of them as draft. Please review other branches as well for similar mistakes

**Comment by AlexMokin** [comment]
indentations are still not consistent, there are missing brackets and semicolon.

Merge the latests changes from master

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js [hunk 13]
```diff
@@ -119,7 +135,18 @@ function isTriangle(a,b,c) {
  *  
  */
 function doRectanglesOverlap(rect1, rect2) {
-    throw new Error('Not implemented');
+    function compare(rec1, rec2){
+       if (
+            rec1.top + rec1.height >= rec2.top &&
+            rec1.left + rec1.width >= rec2.left
+        ) return true
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: missing brackets

File: task/06-conditions-n-loops-tasks.js [hunk 14]
```diff
@@ -119,7 +135,18 @@ function isTriangle(a,b,c) {
...
+        return (
+            rec1.top + rec1.height >= rec2.top &&
+            rec1.left + rec1.width >= rec2.left + rec2.width
+        )
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:160]: missing semicolon

File: task/06-conditions-n-loops-tasks.js [hunk 15]
```diff
@@ -46,7 +49,13 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
-    throw new Error('Not implemented');
+    let i = 1;
+    let fact = 1;
+    while(i < n){
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: missing space after )

File: task/06-conditions-n-loops-tasks.js [hunk 16]
```diff
@@ -296,7 +374,48 @@ function getDigitalRoot(num) {
...
+            }			
+        }
+        if (sum[i] !== 0) return false
+    }
+    for(let i = 0; i < 4; i++){
+        for(let j = 0; j < str.length - 1; j++){
+            if(str[j] === massIn[i]){
+                for(let n = 0; n < 4; n++){
+                    if(massOut[n] === str[j + 1] && str[j + 1] !== massOut[i]){
+                        return false
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: missing semicolon

File: task/06-conditions-n-loops-tasks.js [hunk 17]
```diff
@@ -296,7 +374,48 @@ function getDigitalRoot(num) {
...
+                if(str[j] === massOut[i])sum[i] -= 1;
+            }			
+        }
+        if (sum[i] !== 0) return false
+    }
+    for(let i = 0; i < 4; i++){
+        for(let j = 0; j < str.length - 1; j++){
+            if(str[j] === massIn[i]){
+                for(let n = 0; n < 4; n++){
+                    if(massOut[n] === str[j + 1] && str[j + 1] !== massOut[i]){
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: missing space after )

**Review by AlexMokin — COMMENTED** [review]
there are missing brackets and semicolon, i added comments in several places, but that's not all of them. Find them and fix them. Also double check indentations. Do it for all PRs

Please do not re-open until fixed

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by alexErm0** [comment]
I've put spaces and semicolons in every place I can possibly imagine them in. If my indentations are still incorrect, could you please tell me what kind of problem I have with them, because I've replaced all indetations with 4 spaces and I have no idea, what can be possibly wrong with them now.

File: task/06-conditions-n-loops-tasks.js [hunk 18]
```diff
@@ -373,7 +535,22 @@ function toNaryString(num, n) {
...
+    let flag = "";
+    for (let i = 0; i < pathes[0].length; i++) {
+        flag = "";
+        for (let j = 1; j < pathes.length; j++) {
+            flag = true;
+            flag = (pathes[0].charAt(i) === pathes[j].charAt(i) && flag);
+        }
+        if (flag) {
+            Piece += pathes[0].charAt(i);
+        } else break;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: brackets

File: task/06-conditions-n-loops-tasks.js [hunk 19]
```diff
@@ -373,7 +535,22 @@ function toNaryString(num, n) {
...
+    let Piece = "";
+    let flag = "";
+    for (let i = 0; i < pathes[0].length; i++) {
+        flag = "";
+        for (let j = 1; j < pathes.length; j++) {
+            flag = true;
+            flag = (pathes[0].charAt(i) === pathes[j].charAt(i) && flag);
+        }
+        if (flag) {
+            Piece += pathes[0].charAt(i);
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: naming

File: task/06-conditions-n-loops-tasks.js [hunk 20]
```diff
@@ -356,7 +497,28 @@ function timespanToHumanString(startDate, endDate) {
...
+    let con = 0;
+    while (a <= num) {
+        a *= n;
+        index++;
+    }
+    for (let i = 0; i < index; i++) {
+        a = a / n;
+        if (num < a && i !== 0) {
+            str += "0";
+	} else {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: indentation

File: task/06-conditions-n-loops-tasks.js [hunk 21]
```diff
@@ -296,7 +373,50 @@ function getDigitalRoot(num) {
...
+            if (str[j] === massIn[i]) {
+                sum[i] += 1;
+                if (sum[i] === 1) index[i] = j;
+            }
+        }
+    }
+    for (let i = 0; i < 4; i++) {
+        for (let j = index[i] + 1; j < str.length; j++) {
+            if (sum[i] !== 0) {
+                if (str[j] === massOut[i]) sum[i] -= 1;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: brackets

File: task/06-conditions-n-loops-tasks.js [hunk 22]
```diff
@@ -296,7 +373,50 @@ function getDigitalRoot(num) {
...
+                    return false;
+                }
+            }
+        }
+    }
+    for (let i = 0; i < 4; i++) {
+        for (let j = 0; j < str.length; j++) {
+            if (str[j] === massIn[i]) {
+                sum[i] += 1;
+                if (sum[i] === 1) index[i] = j;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: if you do assigments, even though thisl ine is short still use brackets

File: task/06-conditions-n-loops-tasks.js [hunk 23]
```diff
@@ -270,7 +339,15 @@ function isCreditCardNumber(ccn) {
  *   165536 (1+6+5+5+3+6 = 26,  2+6 = 8) => 8
  */
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    let sum = num;
+    function func(num) {
+        num = new String(num);
+        let result = 0;
+        for (let i = 0; i < num.length; i++) result += new Number(num[i]);
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: 99% of the time it's better to put brackets for loops

**Review by AlexMokin — COMMENTED** [review]
check other prs as well

**Comment by alexErm0** [comment]
I updated all PRs according to your checks.

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**